src/main/java/com/kaytat/simpleprotocolplayer/BufferToAudioTrackThread.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2011 The Android Open Source Project
+ * Copyright (C) 2014 kaytat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.kaytat.simpleprotocolplayer;
+
+import android.media.AudioTrack;
+import android.util.Log;
+
+/**
+ * Worker thread that takes data from the buffer and sends it to audio track
+ */
+class BufferToAudioTrackThread extends WorkerThreadPair.ThreadStoppable {
+    final String TAG;
```

Comment:
it should be. when I touch this code first, most class is member of MusicService, so I think maybe fewer java files  is a goal ? I'll improve that